app/src/main/java/com/duckduckgo/app/settings/SettingsViewModel.kt

```diff
@@ -182,6 +195,19 @@ class SettingsViewModel @Inject constructor(
         )
     }
 
+    fun onFireAnimationSelected(selectedFireAnimation: FireAnimation) {
+        if (settingsDataStore.isCurrentlySelected(selectedFireAnimation)) {
+            Timber.v("User selected same thing they already have set: $selectedFireAnimation; no need to do anything else")
+            return
+        }
+        settingsDataStore.selectedFireAnimation = selectedFireAnimation
+        animationLoader.preloadSelectedAnimation()
```

Comment:
I would have loved to be able to just listen to changes in settings. In that case, animationLoader would just subscribe to whatever is important to thee class, and preload accordingly.
This view model should not be worried about this unrelated business logic (like preloading animations etc.)

This would not necessarily be needed for this PR, but having settings that other classes can subscribe to, simplifies a lot thee logic in general

Thoughts?

Reply:
Yes, I agree with your proposal. Actually, that will be something we will have for free with DataStore (currently in Alpha https://developer.android.com/topic/libraries/architecture/datastore), and most probably I would have done too if the stored value was in Room.

Reply:
Yup, agreed. I wouldn't change it now just to be consistent with the rest of the settings but an overall migration to DataStore at some point would be nice.